### ISSUE TITLE
Determine screen readability of MozFest site

### ISSUE THREAD

**Issue by edrushka** [mozilla/mozillafestival.org#706]
@gvn @alanmoo could either of you please help me determine how screen readable the MozFest website is currently? We are trying to make the entire festival more accessible to people with vision impairments this year. Thank you!!

cc @mmmavis @marcwalsh

**Comment by gvn** [comment]
This is honestly quite subjective and would really be best audited by a dedicated accessibility professional.

However, based on some cursory analysis I'd say that we should change the date and time pickers on [this page](https://mozillafestival.org/fringe) to be native HTML5 form input types for best compatibility with screen readers and assistive input devices. If we don't do that then the libraries should be audited to ensure compatibility.

Generally there's not too much in the way of interactive UI, which is usually what requires special attention to be compatible (eg: carousels).

A good way to test is to see if everything is accessible with just a keyboard. Also using a screen reader app is helpful (OSX has a free voiceover utility).

**Comment by alanmoo** [comment]
We should also use some automated testing to ensure we've got proper alt tags everywhere and some other basic checks. https://github.com/squizlabs/HTML_CodeSniffer

**Comment by mmmavis** [comment]
Thanks @gvn @alanmoo 

Yea I can update the form so it uses native HTML5 inputs plus what @alanmoo has suggested above. @edrushka is there a date that we are thinking about publishing the blog post? We can surely make the site somewhat accessible but from my experience before using screen reader to test will take quite some time (for me at least, since I'm bad at using screen readers) and the work needed really depends on how accessible we want the site to be. Is it fair that we just ensure all nav links are keyboard accessible and content (paragraphs and images) can be picked up by screen readers? Just a note that the site currently has around ~16 pages (mostly static).

**Comment by edrushka** [comment]
Thanks so much @gvn and @alanmoo 🌷 

Mavis, I suggest modifying the blog to indicate we are working on improving the accessibility of the site, rather than waiting for the work to be done, since I think it will be an ongoing process of improvement. 

@hannahkane who do you suggest I talk to about resourcing this work?

**Comment by hannahkane** [comment]
@edrushka - do you mean within our team, or going elsewhere (Moco? a contractor?)

If the former, perhaps @mmmavis @gvn @alanmoo and I can discuss tomorrow (or Thursday since Gavin is out tomorrow) and try to get an estimate on this work.

**Comment by edrushka** [comment]
@hannahkane I'm not sure!

Perhaps a logical first step is a proper audit of our site to see how much is screen readable. With that information, we can prioritize improvements and discuss resourcing. The Guidebook schedule will have to factor in as well, as arguably this is the most important information people need to attend the festival. We can reach out to Guidebook to see what a11y features are already in place.

**Comment by mmmavis** [comment]
(including http://marcysutton.github.io/a11y-automated-testing/#/11 here so I don't lose this link)

@edrushka I'm gonna do a quick pass today and fix obvious a11y flaws if I find any. After that I think it would be great if we could get a few people to help do a more thorough test on each page on the site. I will update my progress by EOD.

/cc @hannahkane @alanmoo

**Comment by mmmavis** [comment]
@edrushka I found this article about [Guidebook and Accessibility](https://support.guidebook.com/hc/en-us/articles/204864754-Guidebook-and-Accessibility-VPAT-Support-) !

**Comment by hannahkane** [comment]
@edrushka - I'm happy to do some testing, once @mmmavis has done her first pass. I'm wondering if you think it would be appropriate to offer this as an opportunity to the other Digital Inclusion wranglers, since we have a special interest in it. Might be a valuable learning opportunity for folks (that's how I'm looking at it!). What do you think?

**Comment by edrushka** [comment]
Yes, if anyone from the Digital Inclusion team is interested in helping, that would be great. 

I do hear @gvn's comment that "This is honestly quite subjective and would really be best audited by a dedicated accessibility professional." and caution us not to be too ad hoc in our approach. Ideally the learnings here could be applied to other MoFo sites as well.

**Comment by mmmavis** [comment]
I did a quick audit with [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/). Most errors we got on the site are colour contrast related. In particular, the hot pink text on white is failing a11y check.

<img width="212" alt="screen shot 2017-08-24 at 9 34 45 am" src="https://user-images.githubusercontent.com/2896608/29677328-a6add514-88af-11e7-8ca6-72ca1ab9edab.png">
<img width="272" alt="screen shot 2017-08-24 at 9 34 42 am" src="https://user-images.githubusercontent.com/2896608/29677327-a6ad20ce-88af-11e7-8aa0-2a9146ace280.png">
<img width="112" alt="screen shot 2017-08-24 at 9 34 33 am" src="https://user-images.githubusercontent.com/2896608/29677326-a6aa4de0-88af-11e7-9f71-6964f9e16f28.png">

I worked with @sabrinang yesterday to fix flaws including updating light colour shades to darker ones and adding missing HTML attributes where needed. We are still having trouble finding a better a11y friendly highlighting colour to replace hot pink (#FA1BAB) at the moment so for that we'll probably need to wait till we have settled down on the colours for MozFest 2017.

@edrushka can you have a quick look on https://mozillafestival-org-sta-pr-713.herokuapp.com/ We changed inline hyperlinks from hot pink to blue and some big headings from blue to body text grey. If this first pass of improvement looks okay I will push these changes live soon.

e.g., compare
- https://mozillafestival-org-sta-pr-713.herokuapp.com/ with https://mozillafestival.org/
- https://mozillafestival-org-sta-pr-713.herokuapp.com/speakers/2016 with https://mozillafestival.org/speakers/2016
- https://mozillafestival-org-sta-pr-713.herokuapp.com/spaces with https://mozillafestival.org/spaces

**Comment by mmmavis** [comment]
I haven't had time to test the site with screen reader yet, but I'm aiming to do that today.

**Comment by edrushka** [comment]
We've just finalized the 2017 palette. Ticket to find a replacement for the hot pink is here: https://github.com/MozillaFoundation/mozfest-design-2017/issues/27
